joomla/joomla-cms
# [4.1] Media webservice

Pull Request for Issue #34224.

### Summary of Changes
This is a followup pr of #34314, where I continue the great work of @pjdevries. I had to change the endpoints slightly as adapters are now also supported. The new endpoints are:

```
GET {{base_path}}/api/index.php/v1/media/adapters
GET {{base_path}}/api/index.php/v1/media/adapters/{id}
GET {{base_path}}/api/index.php/v1/media/files/{path}
POST {{base_path}}/api/index.php/v1/media/files
PATCH {{base_path}}/api/index.php/v1/media/files/{path}
DELETE {{base_path}}/api/index.php/v1/media/files/{path} 
```
~This is in early stage as system tests are still under development which will reveal some more issues along the way.~

### Testing Instructions
Check the system tests. If you want to test it by yourself, see #34314 for testing instructions with curl (don't forget to adapt the endpoints to the new ones mentioned above). Or use postman.
